tests/components/plugwise/test_init.py

```diff
@@ -53,5 +56,40 @@ async def test_unload_entry(hass, mock_smile_adam):
     entry = await async_init_integration(hass, mock_smile_adam)
 
     mock_smile_adam.async_reset = AsyncMock(return_value=True)
-    assert await async_unload_entry(hass, entry)
+    await hass.config_entries.async_unload(entry.entry_id)
+    await hass.async_block_till_done()
+    assert entry.state == ENTRY_STATE_NOT_LOADED
     assert not hass.data[DOMAIN]
+
+
+async def test_unload_gateway_entry(hass, mock_smile_adam):
+    """Test being able to unload an entry."""
+    entry = await async_init_integration(hass, mock_smile_adam)
+
+    mock_smile_adam.async_reset = AsyncMock(return_value=True)
+    assert await plugwise.gateway.async_unload_entry(hass, entry)
+    assert not hass.data[DOMAIN]
+
+
+async def test_async_setup_entry_fail(hass):
+    """Test async_setup_entry."""
+    entry = MockConfigEntry(domain=DOMAIN, data={})
+    config = {}
+
+    await async_setup_component(hass, "plugwise", config)
+    await hass.async_block_till_done()
+    assert not await plugwise.async_setup_entry(hass, entry)
```

Comment:
this should use hass api as well `hass.config_entries.async_setup(entry.entry_id)` and also check the entry state is failed

Reply:
Committed accordingly (though SETUP_FAIL not loading fail).